authors/apps/articles/tests/test_rate_article.py

```diff
@@ -0,0 +1,170 @@
+        self.assertIn(
+            'Authentication credentials were not provided.', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_rate_article(self):
+        """
+        test rate article.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'sucessfully rated', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_rate_article_not_found(self):
+        """
+        test weather rate article without article will fail.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        self.rate_details["user"]["slug"] = "-ss-dd-dd-ff"
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'Article not found', str(
+                response.data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_rate_article_invalid_rate(self):
+        """
+        test rate article with invalid data wil fail.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        self.rate_details["user"]["rate"] = 7
+        response = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.rate_details["user"]["rate"] = 0
+        resp = self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.assertIn(
+            'invalid rate value should be > 0 or <=5', str(
+                response.data))
+        self.assertIn(
+            'invalid rate value should be > 0 or <=5', str(
+                resp.data))
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_get_rate_article(self):
+        """
+        test weather rate article with token.
+        """
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token_2)
+        self.rate_details["user"]['rate'] = 4
+        self.client.post(
+            self.rate_url,
+            self.rate_details,
+            format='json')
+        response = self.client.get(
+            self.view_rates_url + str(1) + "/",
+            format='json')
+        self.assertEqual(
+            3.5,
+            response.data["rates"])
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_rate_article_not_found(self):
+        """
+        test weather get rates.
```

Comment:
typo here: `test whether...`
Provide more context in your docstring as it doesn't make sense even when the typo is fixed.